receiver/k8sclusterreceiver/collection/metadata.go

```diff
@@ -83,17 +83,17 @@ func mergeKubernetesMetadataMaps(maps ...map[ResourceID]*KubernetesMetadata) map
 	return out
 }
 
-// KubernetesMetadataExporter provides an interface to implement
-// ConsumeKubernetesMetadata in Exporters that support metadata.
-type KubernetesMetadataExporter interface {
-	// ConsumeKubernetesMetadata will be invoked every time there's an
-	// update to a resource that results in one or more KubernetesMetadataUpdate.
-	ConsumeKubernetesMetadata(metadata []*KubernetesMetadataUpdate) error
+// MetadataExporter provides an interface to implement
+// ConsumeMetadata in Exporters that support metadata.
+type MetadataExporter interface {
```

Comment:
I think it would make sense to move this interface and the `MetadataUpdate` struct out of this receiver's code base, maybe to `internal/splunk` since this is going to be used by other receivers as well.

Reply:
I would do it as a separate PR. In this one I just want to make sure that signalfx exporter use metadata names instead of kubernetesMetadata. If you ok with merging this, I'll create an issue for moving this interface out

Reply:
Yes, that sounds good.

Reply:
Created a ticket https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/1031